orc8r/cloud/go/services/orchestrator/servicers/protected/indexer_servicer.go

```diff
@@ -117,39 +118,48 @@ func setSecondaryStates(ctx context.Context, networkID string, states state_type
 	if len(sessionIDToIMSI) == 0 && len(cTeidToHwId) == 0 {
 		return stateErrors, nil
 	}
-	multiError := merrors.NewMulti()
+	errs := &multierror.Error{}
 	if len(sessionIDToIMSI) != 0 {
 		err := directoryd.MapSessionIDsToIMSIs(ctx, networkID, sessionIDToIMSI)
-		multiError = multiError.AddFmt(err, "failed to update directoryd mapping of session IDs to IMSIs %+v", sessionIDToIMSI)
+		if err != nil {
+			errs = multierror.Append(errs, errors.Errorf("failed to update directoryd mapping of session IDs to IMSIs %+v %v", sessionIDToIMSI, err))
+		}
 	}
 	if len(cTeidToHwId) != 0 {
 		err := directoryd.MapSgwCTeidToHWID(ctx, networkID, cTeidToHwId)
-		multiError = multiError.AddFmt(err, "failed to update directoryd mapping of control plane teid To HwID %+v", sessionIDToIMSI)
+		if err != nil {
+			errs = multierror.Append(errs, errors.Errorf("failed to update directoryd mapping of control plane teid To HwID %+v %v", sessionIDToIMSI, err))
+		}
 	}
 	if len(uTeidToHwId) != 0 {
 		err := directoryd.MapSgwUTeidToHWID(ctx, networkID, uTeidToHwId)
-		multiError = multiError.AddFmt(err, "failed to update directoryd mapping of user plane teid To HwID %+v", sessionIDToIMSI)
+		if err != nil {
+			errs = multierror.Append(errs, errors.Errorf("failed to update directoryd mapping of user plane teid To HwID %+v %v", sessionIDToIMSI, err))
+		}
 	}
 
-	// multiError will only be nil if both updates succeeded
-	return stateErrors, multiError.AsError()
+	// errs will only be nil if both updates succeeded
+	return stateErrors, errs.ErrorOrNil()
 }
 
 // unsetSecondaryStates removes {sessionID -> IMSI} and {TEID -> HWID} mappings
 func unsetSecondaryStates(ctx context.Context, networkID string, states state_types.StatesByID) (state_types.StateErrors, error) {
 	sessionIDToIMSI, cTeidToHwId, uTeidToHwId, stateErrors := getMappings(states)
-	multiError := merrors.NewMulti()
+	errs := &multierror.Error{}
 
 	err := unsetTeids(ctx, controlPlaneTeid, networkID, cTeidToHwId)
-	multiError = multiError.Add(err)
-
+	if err != nil {
+		errs = multierror.Append(errs, err)
```

Comment:
multierror already ignores `nil`, no need for the if block: https://github.com/hashicorp/go-multierror/blob/9974e9ec57696378079ecc3accd3d6f29401b3a0/append.go#L9